vpr/src/place/initial_placement.cpp

```diff
@@ -51,38 +51,214 @@ static vtr::vector<ClusterBlockId, t_block_score> assign_block_scores();
 //Sort the blocks according to how difficult they are to place, prior to initial placement
 static std::vector<ClusterBlockId> sort_blocks(const vtr::vector<ClusterBlockId, t_block_score>& block_scores);
 
-//Sort the macros according to how difficult they are to place, prior to initial placement
-static std::vector<t_pl_macro> sort_macros(const vtr::vector<ClusterBlockId, t_block_score>& block_scores);
+//Get a legal placement position for the block
+static bool get_legal_placement_loc(PartitionRegion& pr, t_pl_loc& loc, t_logical_block_type_ptr block_type);
+
+static bool try_all_part_region_locations(ClusterBlockId blk_id, PartitionRegion& pr, t_logical_block_type_ptr block_type, enum e_pad_loc_type pad_loc_type);
+
+static bool try_all_part_region_locations_macro(t_pl_macro pl_macro, PartitionRegion& pr, t_logical_block_type_ptr block_type);
 
 void print_sorted_blocks(const std::vector<ClusterBlockId>& sorted_blocks, const vtr::vector<ClusterBlockId, t_block_score>& block_scores);
 
-static int get_free_sub_tile(std::vector<std::vector<int>>& free_locations, int itype, std::vector<int> possible_sub_tiles) {
-    for (int sub_tile : possible_sub_tiles) {
-        if (free_locations[itype][sub_tile] > 0) {
-            return sub_tile;
+static void place_the_blocks(const std::vector<ClusterBlockId>& sorted_blocks,
+                             const vtr::vector<ClusterBlockId, t_block_score>& block_scores,
+                             enum e_pad_loc_type pad_loc_type);
+
+static bool get_legal_placement_loc(PartitionRegion& pr, t_pl_loc& loc, t_logical_block_type_ptr block_type) {
```

Comment:
make pr a const reference, put inputs first and then output

Reply:
Can't make it const, if I do, I can't use the get_partition_region() accessors which I use to get the vector of regions.